From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: CruzAPI <guilhermecruz1574@gmail.com>
Date: Tue, 3 Sep 2024 20:32:48 -0300
Subject: [PATCH] Add GeneratedStructure to AsyncStructureGenerateEvent


diff --git a/src/main/java/org/bukkit/event/world/AsyncStructureGenerateEvent.java b/src/main/java/org/bukkit/event/world/AsyncStructureGenerateEvent.java
index 3ede65e17fb4fb3b1338c5cb470d46f37d576fb1..85b8d8c3221847c03389729b1142f7addc4a69f4 100644
--- a/src/main/java/org/bukkit/event/world/AsyncStructureGenerateEvent.java
+++ b/src/main/java/org/bukkit/event/world/AsyncStructureGenerateEvent.java
@@ -1,15 +1,11 @@
 package org.bukkit.event.world;
 
 import com.google.common.base.Preconditions;
-import java.util.Collections;
-import java.util.LinkedHashMap;
-import java.util.List;
-import java.util.Map;
 import org.bukkit.NamespacedKey;
 import org.bukkit.World;
 import org.bukkit.event.HandlerList;
+import org.bukkit.generator.structure.GeneratedStructure;
 import org.bukkit.generator.structure.Structure;
-import org.bukkit.generator.structure.StructurePiece;
 import org.bukkit.util.BlockTransformer;
 import org.bukkit.util.BoundingBox;
 import org.bukkit.util.EntityTransformer;
@@ -17,6 +13,11 @@ import org.jetbrains.annotations.ApiStatus;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
 /**
  * This event will sometimes fire synchronously, depending on how it was
  * triggered.
@@ -57,8 +58,9 @@ public class AsyncStructureGenerateEvent extends WorldEvent {
     private final Map<NamespacedKey, BlockTransformer> blockTransformers = new LinkedHashMap<>();
     private final Map<NamespacedKey, EntityTransformer> entityTransformers = new LinkedHashMap<>();
     private final List<BoundingBox> pieces;
-
-    public AsyncStructureGenerateEvent(@NotNull World world, boolean async, @NotNull Cause cause, @NotNull Structure structure, @NotNull BoundingBox boundingBox, int chunkX, int chunkZ, List<BoundingBox> pieces) {
+    private final GeneratedStructure generatedStructure;
+    
+    public AsyncStructureGenerateEvent(@NotNull World world, boolean async, @NotNull Cause cause, @NotNull Structure structure, @NotNull BoundingBox boundingBox, int chunkX, int chunkZ, List<BoundingBox> pieces, GeneratedStructure generatedStructure) {
         super(world, async);
         this.structure = structure;
         this.boundingBox = boundingBox;
@@ -66,6 +68,13 @@ public class AsyncStructureGenerateEvent extends WorldEvent {
         this.chunkZ = chunkZ;
         this.cause = cause;
         this.pieces = pieces;
+        this.generatedStructure = generatedStructure;
+    }
+    
+    @NotNull
+    public GeneratedStructure getGeneratedStructure()
+    {
+        return generatedStructure;
     }
     
     public List<BoundingBox> getPieces()
