From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: CruzAPI <guilhermecruz1574@gmail.com>
Date: Mon, 7 Oct 2024 10:03:11 -0300
Subject: [PATCH] Remove clone from getTool() from BlockShearEntityEvent


diff --git a/src/main/java/org/bukkit/event/block/BlockShearEntityEvent.java b/src/main/java/org/bukkit/event/block/BlockShearEntityEvent.java
index 610768bd329b8612627d361fd9a773a7b91ff108..e94a7533ae702d4d930a3e2e48abfc7f9d756951 100644
--- a/src/main/java/org/bukkit/event/block/BlockShearEntityEvent.java
+++ b/src/main/java/org/bukkit/event/block/BlockShearEntityEvent.java
@@ -44,7 +44,7 @@ public class BlockShearEntityEvent extends BlockEvent implements Cancellable {
      */
     @NotNull
     public ItemStack getTool() {
-        return tool.clone();
+        return tool;
     }
 
     @Override
