From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: CruzAPI <guilhermecruz1574@gmail.com>
Date: Mon, 26 Aug 2024 10:22:01 -0300
Subject: [PATCH] Handle level to debit from EnchantItemEvent


diff --git a/src/main/java/net/minecraft/world/inventory/EnchantmentMenu.java b/src/main/java/net/minecraft/world/inventory/EnchantmentMenu.java
index c5e0ca839b31291d92bd580d4c9be78df3c3242c..8e48072a8fa43f5c7ba8b7863a123a7048476d64 100644
--- a/src/main/java/net/minecraft/world/inventory/EnchantmentMenu.java
+++ b/src/main/java/net/minecraft/world/inventory/EnchantmentMenu.java
@@ -279,7 +279,7 @@ public class EnchantmentMenu extends AbstractContainerMenu {
                             itemstack2.enchant(weightedrandomenchant.enchantment, weightedrandomenchant.level);
                         }
 
-                        player.onEnchantmentPerformed(itemstack, j);
+                        player.onEnchantmentPerformed(itemstack, event.getLevelToDebit());
                         // CraftBukkit end
 
                         // CraftBukkit - TODO: let plugins change this
